pennylane_lightning_gpu/src/simulator/StateVectorCudaMPI.hpp

```diff
@@ -0,0 +1,1455 @@
+         }},
+        {"Rot",
+         [&](auto &&wires, auto &&adjoint, auto &&params) {
+             applyRot(std::forward<decltype(wires)>(wires),
+                      std::forward<decltype(adjoint)>(adjoint),
+                      std::forward<decltype(params)>(params));
+         }},
+        {"CRot", [&](auto &&wires, auto &&adjoint, auto &&params) {
+             applyCRot(std::forward<decltype(wires)>(wires),
+                       std::forward<decltype(adjoint)>(adjoint),
+                       std::forward<decltype(params)>(params));
+         }}};
+
+    const std::unordered_map<std::string, custatevecPauli_t> native_gates_{
+        {"RX", CUSTATEVEC_PAULI_X},       {"RY", CUSTATEVEC_PAULI_Y},
+        {"RZ", CUSTATEVEC_PAULI_Z},       {"CRX", CUSTATEVEC_PAULI_X},
+        {"CRY", CUSTATEVEC_PAULI_Y},      {"CRZ", CUSTATEVEC_PAULI_Z},
+        {"Identity", CUSTATEVEC_PAULI_I}, {"I", CUSTATEVEC_PAULI_I}};
+
+    /**
+     * @brief Normalize the index ordering to match PennyLane.
+     *
+     * @tparam IndexType Integer value type.
+     * @param indices Given indices to transform.
+     */
+    template <typename IndexType>
+    inline auto NormalizeIndices(std::vector<IndexType> indices)
+        -> std::vector<IndexType> {
+        std::vector<IndexType> t_indices(std::move(indices));
+        std::transform(t_indices.begin(), t_indices.end(), t_indices.begin(),
+                       [&](IndexType i) -> IndexType {
+                           return BaseType::getNumQubits() - 1 - i;
+                       });
+        return t_indices;
+    }
+
+    void applyCuSVPauliGate(const std::vector<std::string> &pauli_words,
+                            std::vector<int> &ctrls, std::vector<int> &tgts,
+                            Precision param, bool use_adjoint = false) {
+        int nIndexBits = BaseType::getNumQubits();
+
+        cudaDataType_t data_type;
+
+        if constexpr (std::is_same_v<CFP_t, cuDoubleComplex> ||
+                      std::is_same_v<CFP_t, double2>) {
+            data_type = CUDA_C_64F;
+        } else {
+            data_type = CUDA_C_32F;
+        }
+
+        std::vector<custatevecPauli_t> pauli_enums;
+        pauli_enums.reserve(pauli_words.size());
+        for (const auto &pauli_str : pauli_words) {
+            pauli_enums.push_back(native_gates_.at(pauli_str));
+        }
+        const auto local_angle = (use_adjoint) ? param / 2 : -param / 2;
+
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyPauliRotation(
+            /* custatevecHandle_t */ handle_.get(),
+            /* void* */ BaseType::getData(),
+            /* cudaDataType_t */ data_type,
+            /* const uint32_t */ nIndexBits,
+            /* double */ local_angle,
+            /* const custatevecPauli_t* */ pauli_enums.data(),
+            /* const int32_t* */ tgts.data(),
+            /* const uint32_t */ tgts.size(),
+            /* const int32_t* */ ctrls.data(),
```

Comment:
Ok, this is quite OCD, but if you agree we could make `tgts` into `trgts` so it has the same length as `ctrls`. I think I can manage that with a single `sed` if you'd like me to try.

Reply:
aha, let keep it as it is to be aligned with the `StateVectorCudaManaged`. We can change it for both later.

Reply:
Yeah, I meant in the whole repo. You can have a try, in `pennylane_lightning_gpu` type 
```
grep -rl gts | xargs sed -i "s|gts|rgts|g"
```
then see if it compiles.

Reply:
It's just easy because `gts` seems to only come from relevant strings, otherwise it would mess up everything :D